### PR TITLE
Upgrade target sdk to 34

### PR DESCRIPTION
**Pull request by Fabtron**
The following changes have been applied:
Following the [behaviour changes](https://developer.android.com/about/versions/14/behavior-changes-14) from Android 14 we musst apply the following changes:

* [If your app targets Android 14, it must specify appropriate foreground service types](https://developer.android.com/about/versions/14/changes/fgs-types-required). 
  * Since we don't provide any foreground services and request the required permissions for location and camera there are no changes to apply 

* [Restrictions to implicit and pending intents](https://developer.android.com/about/versions/14/behavior-changes-14#safer-intents)
  * Since we use implicit intends to non exported components the package name needs to be inlcuded.  

* [Runtime-registered broadcasts receivers must specify export behavior](https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported)
  * The only broadcast receiver used has been refactored to declare wether it should be exported or not.

There are no more restrictions that we have to cover. 

Pls check we behaviour changes to ensure that we're not missing something.

### CODE REVIEW

File: ui-toolkit/src/main/kotlin/io/snabble/sdk/widgets/snabble/version/ui/VersionWidget.kt [hunk 1]
```diff
@@ -76,7 +76,7 @@ private fun provideAppVersion(context: Context): String {
         @Suppress("DEPRECATION")
         context.packageManager.getPackageInfo(context.packageName, 0)
     }
-    return packageInfo.versionName
+    return packageInfo.versionName.orEmpty()
```
Comment (cmaier) [inline ui-toolkit/src/main/kotlin/io/snabble/sdk/widgets/snabble/version/ui/VersionWidget.kt:79]: Why this change?

**Review by cmaier — CHANGES_REQUESTED** [review]

File: ui-toolkit/src/main/kotlin/io/snabble/sdk/widgets/snabble/version/ui/VersionWidget.kt (hunk 1 above)
Comment (Fabtron) [reply ui-toolkit/src/main/kotlin/io/snabble/sdk/widgets/snabble/version/ui/VersionWidget.kt:79]: The versionname will become nullabe. This change comes with the compile sdk 35. I applied it as i upgraded the compile sdk to 35 and forgot to revert it as i did downgrade it due to some more required changes.

**Review by cmaier — APPROVED** [review]
